CMakeLists.txt

```diff
@@ -365,6 +365,13 @@ IF (KOKKOSKERNELS_ENABLE_TPL_CUBLAS)
 ENDIF()
 IF (KOKKOSKERNELS_ENABLE_TPL_MAGMA)
   LIST(APPEND TPL_LIST "MAGMA")
+  IF (F77_BLAS_MANGLE STREQUAL "(name,NAME) name ## _")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DADD_ -fopenmp -lgfortran")
+  ELSEIF (F77_BLAS_MANGLE STREQUAL "(name,NAME) name ## __")
+    MESSAGE(FATAL_ERROR "MAGMA does not prefer Fortran mangling with double underscore")
```

Comment:
@vqd8a Is this a bug in MAGMA?  If so, have we reported it?  I'm pretty sure they would fix it.  We don't have any apps that use Trilinos and require MAGMA right now, so we can control the MAGMA version (or even patch it if we need to).

Reply:
@mhoemmen Maybe not a bug. In MAGMA users' guide, it only asks for single underscore, uppercase, and no change.
BTW, what should F77_BLAS_MANGLE be for uppercase?

Reply:
@mhoemmen To be consistent with MAGMA documentation, for now, I have updated CMakeLists.txt so that CXX flag will be only set to -DADD_, -DUPCASE, or -DNOCHANGE

Reply:
@vqd8a asked:
> BTW, what should F77_BLAS_MANGLE be for uppercase?

Users of the `F77_BLAS_MANGLE` macro are supposed to specify the function name in both lowercase and uppercase.  That's why it takes two arguments, `name` and `NAME`.  Here is an example of an `extern` declaration using the macro:
```
extern "C" void F77_BLAS_MANGLE(dgemm,DGEMM) ( /* arguments go here */ );
```

Reply:
Thanks @mhoemmen . So, I think what I did in CMakeLists.txt should be correct:
```F77_BLAS_MANGLE```=```(name,NAME) name ## _``` for single underscore (-DADD_)
```F77_BLAS_MANGLE```=```(name,NAME) NAME``` for uppercase (-DUPCASE)
```F77_BLAS_MANGLE```=```(name,NAME) name``` for no change (-DNOCHANGE)

Reply:
@vqd8a Yes, that looks fine.  Thanks!